mountinfo/doc.go

```diff
@@ -0,0 +1,41 @@
+// Package mountinfo provides a set of functions to retrieve information about OS mounts.
+// Currently it supports Linux. For historical reasons, there is also some support for FreeBSD,
+// and a shallow implementation for Windows, but in general this is Linux-only package, so
+// the rest of the document only applies to Linux, unless explicitly specified otherwise.
+//
+// In Linux, information about mounts seen by the current process is available from
+// /proc/self/mountinfo. Note that due to mount namespaces, different processes can
+// see different mounts. A per-process mountinfo table is available from /proc/<PID>/mountinfo,
+// where <PID> is a numerical process identifier.
+//
+// In general, /proc is not a very effective interface, and mountinfo is not an exception.
+// For example, there is no way to get information about a specific mount point (i.e. it
+// is all-or-nothing). This package tries to hide the /proc ineffectiveness by using
+// parse filters while reading mountinfo. A filter can skip some entries, or stop
+// processing the rest of the file once the needed information is found.
+//
+// For any functions that have path as an argument (such as Mounted or various filters),
+// the argument must be an absolute path, with all symlinks resolved, and cleaned.
+// One way to achieve this is to process the path using filepath.EvalSymlinks.
+//
+// NOTE that in many cases there is no need to consult mountinfo at all. Here are some
+// of the cases where mountinfo should not be parsed:
+//
+// 1. Before performing a mount. If you want to check whether a directory is mounted,
+//    call os.Lstat on it and its parent directory, and compare their
+//    st.Sys().(*syscall.Stat_t).Dev fields -- if they differ, then the directory is
+//    the mount point. Optionally, you can also call unix.Statfs and check the Type
+//    field (i.e. filesystem type).
```

Comment:
Would it be useful to have utilities for this? Or too hard to have some thing that's universal enough?

Reply:
I actually plan to change Mounted() to use the above algo, similar to https://github.com/cilium/cilium/pull/12249.

Let me actually do this...

Reply:
https://github.com/moby/sys/pull/20

If we'll merge that one before this, I'll change the text here.

Reply:
Thanks!